### DISCUSSION TITLE
Understanding the linking between data and domain layer.

### DISCUSSION THREAD

**Discussion by amjadalwareh** [android/nowinandroid#574] [Q&A] [answered]
# Hello, I don't know if it is the right place to ask my question.

I just wanted to understand the dependency link between the data layer and the domain layer, 
AFAIK, the data layer should have a link to the domain layer while in NiA it is the opposite.
I can see from `build.gradle.kts :core:domain` the link to the data layer.
```
implementation(project(":core:data"))
```
_Is it a new architecture, you are following in this project and what are the benefits of having it?_

**Comment by lihenggui** [comment]
You can get more information from this link:
https://github.com/android/nowinandroid/blob/main/docs/ArchitectureLearningJourney.md#domain-layer

**Comment by jocoand** [comment, marked as answer]
I have been wondering and still learning about this too, I try to answer based on my understanding so far

- For the benefit, hope this will help
https://developer.android.com/topic/architecture/domain-layer#data-access-restriction

- About dependency flow, yes seems its different from clean architecture, 
https://developer.android.com/topic/architecture#unidirectional-data-flow

- Domain layer https://developer.android.com/topic/architecture/domain-layer
  >> The domain layer is responsible for encapsulating complex business logic, or simple business logic that is reused by multiple ViewModels. This layer is optional because not all apps will have these requirements. 

  Domain layer -> optional & for reusable business logic

- Data layer https://developer.android.com/topic/architecture/data-layer
  >> the data layer contains application data and business logic. 

  Business logic in data layer

As many android developer came from clean architecture before, I think it would be good if there is a section in the architecture guideline that discuss about it.

**Reply by browndali** [reply]
+1 Thanks, for the reference
